### PR TITLE
documentation for /auth /listings and /users

### LINKED ISSUE #97: update api documentation /users endpoints

### LINKED ISSUE #100: update api documentation /listings endpoints

### LINKED ISSUE #119: update api documentation /auth endpoints

### PR DESCRIPTION
**Pull request by lukasherz**
I also did #100 because I forgot, that you would do it. but because it's still on todo in your task list, it shouldn't be a problem, right? just let me know. when later merging your PR, this could be removed

resolves #97, resolves #119, resolves #100

### CODE REVIEW

File: docs/api-v1.json [hunk 1]
```diff
@@ -0,0 +1,1136 @@
...
+          },
+          "message": {
+            "type": "string",
+            "description": "error message",
+            "example": "used a teapot instead of a coffeepot"
+          },
+          "path": {
+            "type": "string",
+            "description": "path of the request that caused the error",
+            "example": "/v1/coffee/"
```
Comment (matthias-imhof) [inline docs/api-v1.json:47]: probably needs to be changed.

File: docs/api-v1.json [hunk 2]
```diff
@@ -0,0 +1,1136 @@
...
+        "properties": {
+          "authentication": {
+            "$ref": "#/components/schemas/AuthenticationData"
+          },
+          "role": {
+            "$ref": "#/components/schemas/Role"
+          },
+          "details": {
+            "$ref": "#/components/schemas/UserDetails"
+          }
```
Comment (matthias-imhof) [inline docs/api-v1.json:135]: shouldn't it also include the publisher?

File: docs/api-v1.json [hunk 3]
```diff
@@ -0,0 +1,1136 @@
...
+            "example": 2500
+          },
+          "rooms": {
+            "type": "number",
+            "description": "number of rooms of the listing",
+            "required": false,
+            "example": 2.5
+          },
+          "publisher": {
+            "$ref": "#/components/schemas/UserGetPublic"
```
Comment (matthias-imhof) [inline docs/api-v1.json:370]: Isn't it only with the UUID of the user

File: docs/api-v1.json [hunk 4]
```diff
@@ -0,0 +1,1136 @@
...
+            "$ref": "#/components/responses/InternalServerError"
+          },
+          "default": {
+            "$ref": "#/components/responses/UndefinedError"
+          }
+        }
+      }
+    },
+    "/v1/users/": {
+      "get": {
```
Comment (matthias-imhof) [inline docs/api-v1.json:770]: /v1/users

File: docs/api-v1.json [hunk 5]
```diff
@@ -0,0 +1,1136 @@
...
+          "500": {
+            "$ref": "#/components/responses/InternalServerError"
+          },
+          "default": {
+            "$ref": "#/components/responses/UndefinedError"
+          }
+        }
+      }
+    },
+    "/v1/listings/": {
```
Comment (matthias-imhof) [inline docs/api-v1.json:956]: same as above

**Review by matthias-imhof — CHANGES_REQUESTED** [review]

File: docs/api-v1.json (hunk 4 above)
Comment (lukasherz) [reply docs/api-v1.json:770]: both works, so it actually doesn't matter

File: docs/api-v1.json (hunk 1 above)
Comment (lukasherz) [reply docs/api-v1.json:47]: why do you think so? its just an example of how an error looks. not a working example

File: docs/api-v1.json (hunk 2 above)
Comment (lukasherz) [reply docs/api-v1.json:135]: the get sends a public user version `private UserGetPublicDto publisher;`

File: docs/api-v1.json (hunk 3 above)
Comment (lukasherz) [reply docs/api-v1.json:370]: the get sends a public user version `private UserGetPublicDto publisher;`

**Review by matthias-imhof — APPROVED** [review]

File: docs/api-v1.json (hunk 2 above)
Comment (matthias-imhof) [reply docs/api-v1.json:135]: yes , the owner of the listing should also be added, no?

**Review by matthias-imhof — CHANGES_REQUESTED** [review]

File: docs/api-v1.json (hunk 2 above)
Comment (lukasherz) [reply docs/api-v1.json:135]: but this is UserPostPut and has nothing to do with listing. I think I just don't get it :(

Comment (matthias-imhof) [reply docs/api-v1.json:135]: oh yes, you're absolutely right. My bad all good!

**Review by matthias-imhof — APPROVED** [review]

**Pull request merged**